src/core/UserIconGroup/UserIconGroup.tsx

```diff
@@ -0,0 +1,133 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+import cx from 'classnames';
+import { UserIcon } from '../UserIcon';
+export type UserIconGroupType = 'default' | 'animated';
+
+const defaultLength = 5;
+const maxLength = 9;
+
+export type UserIconGroupProps = {
+  /**
+   * UserIconGroup group length
+   * @default 5
+   */
+  length?: number;
+  /**
+   * UserIconGroup type
+   * @default 'default'
+   */
+  type?: UserIconGroupType;
+  /**
+   * User Icons in the UserIconGroup.
+   */
+  children: React.ReactNode;
+};
+
+/**
+ * Group User Icons together
+ * @example
+ * <UserIconGroup>
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *   abbreviation="RM"
+ *    backgroundColor="#C8C2B4"
+ *    size="medium"
+ *    title="Robin Mercer"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ * </UserIconGroup>
+ * * @example
+ * <UserIconGroup type='animated'>
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *   abbreviation="RM"
+ *    backgroundColor="#C8C2B4"
+ *    size="medium"
+ *    title="Robin Mercer"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ * </UserIconGroup>
+ */
+export const UserIconGroup = (props: UserIconGroupProps) => {
+  const { children, type = 'default', length = defaultLength } = props;
+
+  const childrenArray = React.Children.toArray(children);
+  const childrenLength = React.Children.count(children);
+  let additionalChildren = '';
+  if (childrenLength > length) {
+    React.Children.forEach(children, (child: JSX.Element, index) => {
+      if (index >= length) {
+        additionalChildren += child.props.title + '\n';
```

Comment:
Hmm... Weird logic

Reply:
Could you elaborate?

Reply:
Well, I was going to, but then I thought it would be better to scrap this whole block and rethink the API.

If we allow passing `onClick`/`onMouseOver` handlers, then we won't need any of this tooltip logic.